File: Axiom/Internal/Ast/AssignmentExpression.cs
namespace Axiom.Internal.Ast
{
    internal class AssignmentExpression : Expression
    {
        public AssignmentExpression(Token assignmentOperator, Expression left, Expression right)
        {
            Operator = assignmentOperator;
            Destination = left;
            Source = right;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public Expression Destination { get; private set; }
        public Expression Source { get; private set; }
        public Token Operator { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/AssignmentExpression.java
package com.moybl.axiom.ast;

import com.moybl.axiom.Position;
import com.moybl.axiom.Token;

public class AssignmentExpression extends Expression {

	private Token operator;
	private Expression source;
	private Expression destination;

	public AssignmentExpression(Position position, Token operator, Expression source, Expression destination) {
		super(position);

		this.operator = operator;
		this.source = source;
		this.destination = destination;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public Token getOperator() {
		return operator;
	}

	public Expression getSource() {
		return source;
	}

	public Expression getDestination() {
		return destination;
	}

}
